src/main/java/io/vertx/core/json/JsonObject.java

```diff
@@ -70,6 +70,32 @@ public JsonObject(Map<String, Object> map) {
   }
 
   /**
+   * Create a JsonObject from the fields of a Java object.
+   * Faster than calling `new JsonObject(Json.encode(pojo))`.
+   * 
+   * @param obj
+   *          The object to convert to a JsonObject.
+   * @throws IllegalArgumentException
+   *          if conversion fails due to an incompatible type.
+   */
+  @SuppressWarnings("unchecked")
+  public JsonObject(Object obj) {
```

Comment:
Maybe then there should be a static factory method `JsonObject.fromInstance(Object)`, to match `JsonObject#toInstance()`? It would be nice to only ever have to deal with JsonObject, and not additionally with Json.